File: HACKATHON/js/incomeExpense.js
// Income & Expense Tracker
function addTransaction(type, amount, category) {
  let transactions = JSON.parse(localStorage.getItem("transactions")) || [];
  transactions.push({ type, amount, category, date: new Date().toLocaleDateString() });
  localStorage.setItem("transactions", JSON.stringify(transactions));
}

function getTransactions() {
  return JSON.parse(localStorage.getItem("transactions")) || [];
}

function showTransactions() {
  let list = getTransactions();
  let output = "<h3>Transactions</h3><ul>";
  list.forEach(t => {
    output += `<li>${t.date} - ${t.type}: ₹${t.amount} (${t.category})</li>`;
  });
  output += "</ul>";
  document.getElementById("content").innerHTML = output;
}

File: HACKATHON/java/IncomeExpense.java
public class IncomeExpense {
    private String type;   // Income or Expense
    private double amount;
    private String category;
    private String date;

    public IncomeExpense(String type, double amount, String category, String date) {
        this.type = type;
        this.amount = amount;
        this.category = category;
        this.date = date;
    }

    public double getAmount() { return amount; }
    public String getType() { return type; }
    public String getCategory() { return category; }
    public String getDate() { return date; }
}

